native/arrow_format_nif/src/message/schema/field.rs

```diff
@@ -0,0 +1,42 @@
+use crate::utils::CustomMetadata;
+use rustler::{Atom, Decoder, Encoder, NifRecord, Term};
+
+#[derive(Debug, NifRecord)]
+#[tag = "field"]
+pub struct Field {
+    pub name: Name,
+    pub nullable: bool,
+    pub r#type: Atom,
+    pub dictionary: Atom,
+    pub children: Vec<Field>,
+    pub custom_metadata: CustomMetadata,
+}
+
+// TODO: Update Option<T>'s Encoder and Decoder implementations to treat
+// undefined as None in rustler main.
+#[derive(Debug)]
+pub enum Name {
+    Atom(Atom),
+    String(String),
+}
+
+impl Encoder for Name {
+    fn encode<'a>(&self, env: rustler::env::Env<'a>) -> Term<'a> {
+        match &self {
+            Name::Atom(_) => crate::atoms::undefined().encode(env),
+            Name::String(name) => name.clone().into_bytes().encode(env),
```

Comment:
Do you need .clone() if you're doing .into_bytes()?

Reply:
The code fails to compile without cloning.
I'll paste the error when I'm free.

Reply:
```
$ cargo b
   Compiling arrow_format_nif v0.1.0 (/home/ben/Github/serde_arrow/native/arrow_format_nif)
error[E0507]: cannot move out of `*name` which is behind a shared reference
   --> src/message/schema/field.rs:27:35
    |
27  |             Name::String(name) => name.into_bytes().encode(env),
    |                                   ^^^^^------------
    |                                   |    |
    |                                   |    `*name` moved due to this method call
    |                                   move occurs because `*name` has type `String`, which does not implement the `Copy` trait
    |
note: this function takes ownership of the receiver `self`, which moves `*name`
   --> /home/ben/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/string.rs:867:23
    |
867 |     pub fn into_bytes(self) -> Vec<u8> {
    |                       ^^^^

For more information about this error, try `rustc --explain E0507`.
error: could not compile `arrow_format_nif` due to previous error
```